nebula-logger/plugins/big-object-archiving/plugin/classes/LogEntryArchivePlugin.cls

```diff
@@ -0,0 +1,225 @@
+//------------------------------------------------------------------------------------------------//
+// This file is part of the Nebula Logger project, released under the MIT License.                //
+// See LICENSE file or go to https://github.com/jongpie/NebulaLogger for full license details.    //
+//------------------------------------------------------------------------------------------------//
+
+/**
+ * @group Plugins
+ * @description Optional plugin that provides a BigObject, `LogEntryArchive__b`, as an alternative option
+ *              to the platform event `LogEntryEvent__e`
+ * @see LoggerPlugin
+ */
+public without sharing class LogEntryArchivePlugin implements LoggerPlugin.Batchable, LoggerPlugin.Triggerable {
+    @TestVisible
+    private static final String BIG_OBJECT_IMMEDIATE_LOGGER_SAVE_METHOD = LoggerParameter.getString('CustomSaveMethodBigObjectImmediate', null);
+    @TestVisible
+    private static final String BIG_OBJECT_QUEUEABLE_LOGGER_SAVE_METHOD = LoggerParameter.getString('CustomSaveMethodBigObjectQueueable', null);
+    @TestVisible
+    private static final String BIG_OBJECT_LOG_PURGE_ACTION = LoggerParameter.getString('CustomLogPurgeActionArchive', null);
+    @TestVisible
+    private static final String BIG_OBJECT_STORAGE_LOCATION = LoggerParameter.getString('CustomStorageLocationBigObject', null);
+    @TestVisible
+    private static final List<LogEntryArchive__b> LOG_ENTRY_ARCHIVES_TO_SAVE = new List<LogEntryArchive__b>();
+
+    @TestVisible
+    private static String lastSaveMethodUsed;
+
+    /**
+     * @description Default constructor
+     */
+    @SuppressWarnings('PMD.EmptyStatementBlock')
+    public LogEntryArchivePlugin() {
+    }
+
+    // Batchable interface methods
+    /**
+     * @description Skips directly deleting `LogEntryTag__c` records in `LogBatchPurger` so that the tags
+     *             can be included when `LogEntry__c` records are archived into `LogEntryArchive__b`
+     * @param  configuration The instance of `LoggerPlugin__mdt` configured for this specific plugin
+     * @param  input The instance of `LoggerBatchableContext`, provided by the logging system
+     */
+    public void start(LoggerPlugin__mdt configuration, LoggerBatchableContext input) {
+        // Skip directly deleting LogEntryTag__c records so that the tags can be included when LogEntry__c records
+        // are archived into LogEntryArchive__b
+        if (input.sobjectType == Schema.LogEntryTag__c.SObjectType) {
```

Comment:
I feel like it makes more sense to do `!= Schema.LogEntry__c.SObjectType` here - comparing to `LogEntryTag__c` feels like this class knows a bit too much